lib/cdo/aws/s3_packaging.rb

```diff
@@ -149,8 +149,14 @@ def create_package(sub_path)
 
     @logger.info "Attempting to download: #{s3_key}\nto #{package.path}"
     File.open(package, 'wb') do |file|
-      @client.get_object(bucket: BUCKET_NAME, key: s3_key) do |chunk|
-        file.write(chunk)
+      begin
+        @client.get_object(bucket: BUCKET_NAME, key: s3_key) do |chunk|
+          file.write(chunk)
+        end
+      rescue Aws::Errors::MissingCredentialsError, Aws::S3::Errors::ServiceError
+        # Fallback to public URL download if credentials are not provided or invalid.
+        url = Aws::S3::Bucket.new(BUCKET_NAME).object(s3_key).public_url
+        IO.copy_stream open(url), file
```

Comment:
No downside, I can make that change. Actually, working on this I realized we need to make sure the interface remains consistent when using `open(url)` so it continues to throw NoSuchKey when a file is not found. I'll update the PR to make sure this is the case.